.claude/hooks/post-go-lint.sh

```
@@ -0,0 +1,39 @@
+# Uses go run like the Makefile — no global install required, version pinned via Makefile.
+set -uo pipefail
+
+input=$(cat)
+file_path=$(echo "$input" | python3 -c \
+  "import sys,json; d=json.load(sys.stdin); print(d.get('tool_input',{}).get('file_path',''))" \
+  2>/dev/null || echo "")
+
+[[ "$file_path" == *.go ]] || exit 0
+
+REPO_ROOT=$(git -C "$(dirname "$file_path")" rev-parse --show-toplevel 2>/dev/null) || exit 0
+cd "$REPO_ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)